src/main/com/example011/radek/v1/Mines.java

```diff
@@ -0,0 +1,138 @@
+package main.com.example011.radek.v1;
+
+import java.security.NoSuchAlgorithmException;
+import java.security.SecureRandom;
+import java.util.Scanner;
+
+public class Mines {
+
+    protected int playgroundSize;
+
+    protected int numberOfMines;
+
+    protected String[][] playground;
+
+    public Mines(int size, int mines) {
+
+        if (checkPlaygroundSize(size)) {
+            setPlaygroundSize(size);
+            setNumberOfMines(mines);
+        } else {
+            System.out.println("Bad size of playground, size must be bigger then 5!");
+            System.exit(0);
+        }
+    }
+
+    public void fillPlaygroundWithMines() throws NoSuchAlgorithmException {
+
+        playground = new String[playgroundSize][playgroundSize];
+        SecureRandom random = SecureRandom.getInstanceStrong();
+        int placedMines = 0;
+        while (placedMines != getNumberOfMines() || placedMines == 0) {
+            int generatedRowNumber = random.nextInt(playgroundSize);
+            int generatedColumnNumber = random.nextInt(playgroundSize);
+            if (playground[generatedRowNumber][generatedColumnNumber] == null) {
+                playground[generatedRowNumber][generatedColumnNumber] = StringConstant.MINE.getConstant();
+                placedMines++;
+            }
+        }
+    }
+
+    public void printPlayground(boolean showMines) {
+
+        System.out.println("Generating playground...");
+        StringBuilder playingField = new StringBuilder(playgroundSize * playgroundSize * 3);
+        for (int i = 0; i < playgroundSize; i++) {
+            for (int j = 0; j < playgroundSize; j++) {
+                if (playground[i][j] != null && ((showMines && playground[i][j].equals("x")) || playground[i][j].equals("o"))) {
+                    playingField.append(playground[i][j])
+                            .append(StringConstant.TABULATOR.getConstant());
+                } else {
+                    playingField.append(StringConstant.EMPTY.getConstant())
+                            .append(StringConstant.TABULATOR.getConstant());
+                }
+            }
+            playingField.append(StringConstant.NEW_LINE.getConstant());
+        }
+        System.out.println(playingField);
+    }
+
+    public void getSizeAndMines() {
+
+        System.out.println("Playground size is: " + playgroundSize + " and on this ground you can find: " + numberOfMines + " mines.");
+    }
+
+    private void setPlaygroundSize(int size) {
+
+        playgroundSize = size;
+    }
+
+    private boolean checkPlaygroundSize(int size) {
+
+        return size >= 5;
+    }
+
+    private int getPlaygroundSize() {
+
+        return playgroundSize;
+    }
+
+    private void setNumberOfMines(int mines) {
+
+        int size = getPlaygroundSize();
+        if (mines >= (size * size)) {
+            System.out.println("You can't have more mines, then you have playground size !");
+            System.exit(0);
+        } else if ((size * size) < 10) {
```

Comment:
1) magic number pryc ... kontanta a nazve neco jako MIN_COUNT_FOR_ONE_MINE
2) vyrazy v ifech nahradit metodou, kterou pekne pojmenujes, potom podle nazvu metody, ktera se bude v ifu volat hned uvidime, proc do toho bloku skoci
3) tobe tolik ifu urcite nevadi, ale v mych ocich uz to neni vubec pekne a chtelo by to prepsat, nicmene to neprepisuj, ale potom, jak budes na tom lip, tak se snaz hodne ifum vyhnout ...
https://www.baeldung.com/java-replace-if-statements

Reply:
1. Hotovo
2. Hotovo
3. Ok, jak koukám možností je docela dost, switche nejsou problém, ale ty se mi nehodí když je vše rozděleno do metod. Zkusím se k tomu vrátit později, až budu umět více